worker/uniter/operation/runhook.go

```diff
@@ -174,15 +180,25 @@ func (rh *runHook) afterHook(state State) (bool, error) {
 		logger.Debugf("unit %v has started but has not yet set status", ctx.UnitName())
 		// We've finished the start hook and the charm has not updated its
 		// own status so we'll set it to unknown.
-		err = rh.runner.Context().SetUnitStatus(jujuc.StatusInfo{
+		err = ctx.SetUnitStatus(jujuc.StatusInfo{
 			Status: string(status.Unknown),
 		})
+	case hooks.RelationBroken:
+		var isLeader bool
+		isLeader, err = ctx.IsLeader()
+		rel, err := ctx.Relation(rh.info.RelationId)
+		if err != nil {
+			return false, err
+		}
+		suspended, err := rel.Suspended()
+		if err != nil {
+			return false, err
+		}
+		if suspended && isLeader {
+			err = rel.SetStatus(relation.Suspended)
+		}
 	}
-	if err != nil {
-		logger.Errorf("error updating workload status after %v hook: %v", rh.info.Kind, err)
-		return false, err
-	}
-	return hasRunStatusSet, nil
+	return hasRunStatusSet && err == nil, err
```

Comment:
what's the point of this change? the first value should be ignored if err != nil anyway

Reply:
the convention is normally to return the "zero" value if err != nil